Fliglio/Chinchilla/Connection.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Fliglio\Chinchilla;
+
+use PhpAmqpLib\Connection\AMQPConnection;
+
+// Supports multiple connections via configure([]);
+class Connection {
+
+	private static $configs = [];
+	private static $connections = [];
+
+	public static function configure(array $configs) {
+		foreach ($configs as $config) {
+			if (!$config->getName()) {
+				throw new ConnectionException("Name must be set and uniqie for multiple connections.");
+			}
+			self::addConfig($config);
+		}
+	}
+
+	public static function addConfig(Config $config) {
+		if (isset(self::$configs[$config->getName()])) {
+			throw new ConnectionException(sprintf("Config[%s] already set.", $config->getName()));
+		}
+
+		self::$configs[$config->getName()] = $config;
+	}
+
+	public static function hasConnection() {
+		return count(self::$connections) > 0;
+	}
+
+	public static function hasConfigs() {
+		return count(self::$configs) > 0;
+	}
+
+	public static function get($name = null) {
```

Comment:
how pissed would you be if we didn't handle this as a registry/factory, but instead injected it from the application configs into the clients / resources?

(not sure we should, but we should discuss it)